storage/internal/benchmarks/warmup.go

```diff
@@ -66,7 +66,6 @@ func discardBenchmarkResults(ctx context.Context) {
 		for {
 			select {
 			case <-ctx.Done():
```

Comment:
It seems like it's possible that both cases are true at the same time and we randomly select the first case. In this situation, there might still be results to read from the channel, which we'd need to flush -- either here, or before we begin the main benchmarking run. Does this seem accurate?

Reply:
In such a case, `results = make(chan benchmarkResult, 100)` in the main benchmarking recording code will discard any results queued on the old channel. The race condition that was happening was that `make` was happening before the `case <-ctx.Done` was, and that was closing the newly made channel. 
The bigger problem here would be the second case being selected, though it is extremely unlikely as this routine would need to not run until an object is written and the throughput is sent to the result channel. However, since it is technically possible, I added some lines here to deal with it.

Reply:
As discussed, I removed some of the async logic to make this easier to understand. Race condition should be fixed in any case